### PR TITLE
!

### PR DESCRIPTION
**Pull request by PaltineanuRares956**

### CODE REVIEW

File: Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        {
+            return 0;
+        }
+
+    }
+    return 1;
+}
+
+///firstly i write the function like this: void convert(char hex[200], double &a)
+///but the error is saying "expected ';', ',', or ')' before '&' token and idk what is meaning
```
Comment (Alex-Cosma) [inline Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c:21]: Well, if you want to pass an address as a parameter, you should be using the `*` operator, like:
` int convert(char *hex, double *a)`
in this case, `char *hex` is equivalent to `char hex[200]`
But I'm not particularly sure what you wanted to write there

File: Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+int convert(char hex[200], double a)
+{
+    int b = strlen(hex);
+    for(int i = 0; i < b; i++)
+    {
+        switch(hex[i])
+        {
+        case '1':
+            a = a + (hex[i] - '0')*(pow(16,strlen(hex) - i - 1));
```
Comment (Alex-Cosma) [inline Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c:31]: These lines are all the same, I'm not sure if `C` compilers will let you write what I'm about to show you, but you can try:

```
case '1':
case '2':
case '3':
case '4':
case '5':
case '6':
case '7':
case '8':
case '9':
    a = a + (hex[i] - '0')*(pow(16,strlen(hex) - i - 1));
    break
```

Comment (Alex-Cosma) [reply Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c:31]: Again, it might not work in C but it's worth the shot

**Review by Alex-Cosma — COMMENTED** [review]
Nice work

File: Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c (hunk 2 above)
Comment (PaltineanuRares956) [reply Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c]: I tried and it worked!

File: Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c (hunk 1 above)
Comment (PaltineanuRares956) [reply Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c]: I just wanted to modify the address of a in the function

Comment (PaltineanuRares956) [reply Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c:21]: I wanted to modify the address of a in the function

File: Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c (hunk 2 above)
Comment (PaltineanuRares956) [reply Paltineanu Rares-Mihai/3.7 - curs4 switch/main.c:31]: i tried and it worked

**Pull request merged**